pkg/libinit/root_linux_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package libinit
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+)
+
+func TestLoadModule(t *testing.T) {
+	var loadedModules []string
+	loader := &InitModuleLoader{
+		Cmdline: cmdline.NewCmdLine(),
+		Prober: func(name, params string) error {
+			loadedModules = append(loadedModules, name)
+			return nil
+		},
+	}
+
+	expectedModules := []string{"test", "something-test"}
+	InstallModules(loader, expectedModules)
+	if !reflect.DeepEqual(expectedModules, loadedModules) {
```

Comment:
(here and below) if you use `cmp.Diff` (https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff) you get a nice diff, instead of 2 long lists that you have to manually diff

```
if diff := cmp.Diff(expectedModules, loadedModules); diff != "" {
  t.Fatalf("unexpected difference of loaded modules (-want, +got): %v", diff)
}
```